File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/PermissionRequest.java
package za.co.bwmuller.easypermissions;

import java.util.ArrayList;
import java.util.Random;

/**
 * Created by Bernhard Müller on 10/14/2016.
 */
public class PermissionRequest {
    private static Random random;
    private ArrayList<String> permissions;
    private int requestCode;
    private PermissionCallback permissionCallback;

    public PermissionRequest(int requestCode) {
        this.requestCode = requestCode;
    }

    public PermissionRequest(ArrayList<String> permissions, PermissionCallback permissionCallback) {
        this.permissions = permissions;
        this.permissionCallback = permissionCallback;
        if (random == null) {
            random = new Random();
        }
        this.requestCode = random.nextInt(255);
    }

    @Override
    public int hashCode() {
        return requestCode;
    }

    public boolean equals(Object object) {
        if (object == null) {
            return false;
        }
        return object instanceof PermissionRequest && ((PermissionRequest) object).requestCode == this.requestCode;
    }

    public ArrayList<String> getPermissions() {
        return permissions;
    }

    public int getRequestCode() {
        return requestCode;
    }

    public PermissionCallback getPermissionCallback() {
        return permissionCallback;
    }
}

File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/PermissionRequest.kt
package za.co.bwmuller.easypermissions

/**
 * Created by Bernhard Müller on 10/14/2016.
 */
class PermissionRequest {

    val permissions: List<String>
    val requestCode: Int
    val permissionCallback: PermissionCallback?

    constructor(requestCode: Int) {
        permissions = emptyList()
        this.requestCode = requestCode
        permissionCallback = null
    }

    constructor(permissions: List<String>, permissionCallback: PermissionCallback?) {
        this.permissions = permissions
        this.permissionCallback = permissionCallback
        requestCode = 5000 + (System.currentTimeMillis().toInt() % 60000)
    }

    override fun hashCode(): Int {
        return requestCode
    }

    override fun equals(other: Any?): Boolean {
        return if (other == null) {
            false
        } else other is PermissionRequest && other.requestCode == requestCode
    }
}